### PR TITLE
[WIP] model: tensorflow: Refactor tensorflow model

### PR DESCRIPTION
**Pull request by sk-ip**
references #383 
the tensorflow model will be refactored to improve code reusability and make things more neat.

### CODE REVIEW

File: model/tensorflow/dffml_model_tensorflow/dnnc.py [hunk 1]
```diff
@@ -23,6 +24,24 @@
 from dffml.model.model import ModelContext, Model, ModelNotTrained
 
 
+@dataclass
+class BaseConfig:
```
Comment (pdxjohnny) [inline model/tensorflow/dffml_model_tensorflow/dnnc.py:28]: `BaseConfig` is used other places, we should probably name it something more specific

```suggestion
class TensorflowBaseConfig:
```

**Comment by sk-ip** [comment]
Hello @pdxjohnny , thanks for pointing that out, I was thinking of moving the `accuracy` and `predict` to `TensorflowModelContext`? But I am not able to find how to do it. I was thinking maybe a decorator will help but, I am not sure how, can you help me with this.

**Comment by sk-ip** [comment]
@pdxjohnny , i moved some of the common code in `predict` method to `TensorflowModelContext`. and regarding the `accuracy` method, i did not feel it needed refactoring, so its left untouched. please let me know of any changes you may want : )

File: CHANGELOG.md [hunk 2]
```diff
@@ -31,6 +31,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Windows support by selecting `asyncio.ProactorEventLoop` and not using
   `asyncio.FastChildWatcher`.
 - Moved SLR into the main dffml package and removed `scratch:slr`.
+- Refactor `model/tensroflow`
```
Comment (pdxjohnny) [inline CHANGELOG.md:34]:
```suggestion
### Changed
- Refactor `model/tensroflow`
```

**Comment by pdxjohnny** [comment]
Nice work @sk-ip!! Thanks for cleaning things up in there! :)

**Pull request merged**